puzzles/templates/puzzles_table.html

```diff
@@ -75,30 +76,35 @@
 
 <script type="text/javascript">
 $(document).ready(function() {
+    var puzzle_table = $('#puzzles')
+    var puzzle_table_dom = puzzle_table.get(0);
 
     var start_ts = Date.now();
     // For choice fields, each entry should only be searchable by the currently
     // selected value.
-    $.fn.dataTableExt.ofnSearch['select-input'] = function(value) {
-        return $(value).find('select').val();
+    $.fn.dataTableExt.ofnSearch['select-input'] = function(i, j) {
+        var cell = puzzle_table_dom.rows[i + 1].cells[j];
+        return cell.querySelector('select').value;
     };
 
     // For tags, each entry should be searchable by existing badges.
-    $.fn.dataTableExt.ofnSearch['tag-input'] = function(value) {
+    $.fn.dataTableExt.ofnSearch['tag-input'] = function(i, j) {
         var search_str = "";
-        $(value).find(".badge").each( function () {
-            search_str += $(this).text()
-        } );
+        puzzle_table_dom.rows[i + 1].cells[j].querySelectorAll(".badge")
+            .forEach( function (el) {
+                search_str += el.textContent;
+            });
         return search_str;
     };
 
     // Searchable string should contain the title of the puzzle, concatenated
     // with "meta" if it is a metapuzzle (so we can filter this column by
     // metas).
-    $.fn.dataTableExt.ofnSearch['name-input'] = function(value) {
-        var search_str = ""
-        search_str += $(value).find(".puzzle-title").text();
-        search_str += $(value).find(".badge-meta").text();
+    $.fn.dataTableExt.ofnSearch['name-input'] = function(i, j) {
```

Comment:
Is this some modification you made to DataTables to return row/col indices instead of the cell value?

I'm slightly concerned about how maintainable our custom modifications to DataTables are. It would be good to include a diff of your DataTables modifications for reference.

Reply:
Yeah, see the change to jQuery.dataTables.js in https://github.com/cardinalitypuzzles/smallboard/pull/113/commits/c6ded59df98d408445a6b8af33dad9a08ac9ebe0#diff-ae57b08d2f5ce76bf117f38f28a88571R4503

It's pretty trivial; I just pass i and j instead of the data that dataTable is storing. This possibly becomes obsolete if you render the table from JSON, at which point we can delete our fork and use the CDN dataTables.